configure.ac

```diff
@@ -638,6 +638,7 @@ dnl Define cpp macros in CPPFLAGS, rather than doing AC_DEFINE(macro), since the
 dnl definitions need to be seen before any headers are included, which is a pain
 dnl to make happen otherwise.
 default_retain="0"
+zero_realloc_default_alloc="0"
```

Comment:
I might invert the direction of this test; assume it's `alloc` by default, and change to `free` only if we know it's the platform behavior

(e.g. on gnu linux it should be `free`, on musl linux it should be `alloc`; defaulting to the safe-but-leaky choice).

Reply:
Updated